### PR TITLE
BUG: GH10365 in interpolate_1d when method is piecewise_polynomial

### PR DESCRIPTION
**Pull request by kawochen**
Starting point for closing #10365.
I only have the trivial case. I think this method (and perhaps other methods as well) needs #10383 for it to be useful.

### CODE REVIEW

File: pandas/tests/test_generic.py [hunk 1]
```diff
@@ -872,6 +872,14 @@ def test_interp_limit_no_nans(self):
         expected = s
         assert_series_equal(result, expected)
 
+    def test_interp_piecewise_polynomial(self):
+        # GH 10365
+        tm._skip_if_no_scipy()
+        s = Series([1, 2, 3, 4, np.nan, 6, np.nan])
+        result = s.interpolate(method='piecewise_polynomial')
+        expected = Series([1., 2., 3., 4., 5., 6., 7.])
+        assert_series_equal(expected, result)
```
Comment (jreback) [inline pandas/tests/test_generic.py]: can you add some tests with nans up front? (maybe for several other interpolate methods as well to make sure nothing breaking)

**Comment by kawochen** [comment]
Added tests for `krogh` and `barycentric`.

**Comment by jreback** [comment]
can you rebase

**Comment by kawochen** [comment]
Maybe I should just keep the tests for krogh and barbycentric? Piecewise polynomial needs higher order derivatives to be passed in for it to be useful -- otherwise it's just linear interpolation; not sure what a good way would be.

**Comment by jreback** [comment]
@kawochen what's the status of this?

**Comment by kawochen** [comment]
Still don't have a good way to pass in the higher order derivatives.  Will close if for now.

**Pull request closed without merging**